3.6/config.sh

```diff
@@ -23,3 +23,12 @@ if [ ! -z "$POSTGRES_SYNCHRONOUS_COMMIT" ]; then sed -i "s/synchronous_commit =
 if [ ! -z "$POSTGRES_MAX_WAL_SIZE" ]; then sed -i "s/max_wal_size = 1GB/max_wal_size = $POSTGRES_MAX_WAL_SIZE/g" /etc/postgresql/12/main/conf.d/postgres-tuning.conf; fi
 if [ ! -z "$POSTGRES_CHECKPOINT_TIMEOUT" ]; then sed -i "s/checkpoint_timeout = 10min/checkpoint_timeout = $POSTGRES_CHECKPOINT_TIMEOUT/g" /etc/postgresql/12/main/conf.d/postgres-tuning.conf; fi
 if [ ! -z "$POSTGRES_CHECKPOINT_COMPLETITION_TARGET" ]; then sed -i "s/checkpoint_completion_target = 0.9/checkpoint_completion_target = $POSTGRES_CHECKPOINT_COMPLETITION_TARGET/g" /etc/postgresql/12/main/conf.d/postgres-tuning.conf; fi
+
+
+# import style tuning
+if [ "$IMPORT_STYLE" = "" ]; then
+    echo "Use full import style"
+else
+    echo "Use ${IMPORT_STYLE} import style"
+    printf "@define('CONST_Import_Style', CONST_BasePath.'/settings/import-${IMPORT_STYLE}.style');" >> /app/src/build/settings/local.php
```

Comment:
Aren't you adding a line to the end of the file every time that the container starts?

Reply:
Yes, for some reason, this line is not present in the running container local.php so the replacement doesn't work.

```bash
docker exec -i -t nominatim /bin/bash
root@980b1af106f3:/app# cat src/build/settings/local.php
<?php
 // Paths
 @define('CONST_Postgresql_Version', '12');
 @define('CONST_Postgis_Version', '3');
 // Website settings
 @define('CONST_Website_BaseURL', '/');
 @define('CONST_Replication_Url', 'https://download.geofabrik.de/europe/monaco-updates/');
 @define('CONST_Replication_MaxInterval', '86400');     // Process each update separately, osmosis cannot merge multiple updates
 @define('CONST_Replication_Update_Interval', '86400');  // How often upstream publishes diffs
 @define('CONST_Replication_Recheck_Interval', '900');   // How long to sleep if no update found yet
 @define('CONST_Pyosmium_Binary', '/usr/local/bin/pyosmium-get-changes');
 //@define('CONST_Database_DSN', 'pgsql:host=192.168.1.128;port=6432;user=nominatim;password=password1234;dbname=nominatim'); <driver>:host=<host>;port=<port>;user=<username>;password=<password>;dbname=<database>
```

I'm puzzled by this, the Replication_URL changes works but none of my changes.

Reply:
First try was to add it to `\nominatim-docker\3.6\conf.d\local.php` but the line is still absent from the running one.

Reply:
Once more: are you building a local container image with `docker build .` before executing it (takes about 3-5 minutes, so you will notice)? If you use the image `mediagis/nominatim:3.6` it will be the one downloaded from the internet, not the one that you are editing.

Reply:
The script I'm using:

#Stop the nominatim container
docker stop nominatim
```bash
# clean stopped containers
docker container prune

# change directory to docker folder
cd ~/nominatim-docker/3.6/

# build / update the image if needed
docker build --pull --rm -t nominatim .

# version without data persistence

docker run -it --rm --shm-size=8G \
  -e PBF_URL=https://download.geofabrik.de/europe/monaco-latest.osm.pbf \
  -e REPLICATION_URL=https://download.geofabrik.de/europe/monaco-updates/ \
  -e IMPORT_WIKIPEDIA=false \
  -e NOMINATIM_PASSWORD=very_secure_password \
  -e THREADS=6 \
  -e POSTGRES_MAX_WAL_SIZE=3GB \
  -e IMPORT_STYLE=admin \
  -p 8080:8080 \
  --name nominatim \
  mediagis/nominatim:3.6

```

Reply:
You're almost there!

The last command should be:

```
docker run -it --rm --shm-size=8G \
  -e PBF_URL=https://download.geofabrik.de/europe/monaco-latest.osm.pbf \
  -e REPLICATION_URL=https://download.geofabrik.de/europe/monaco-updates/ \
  -e IMPORT_WIKIPEDIA=false \
  -e NOMINATIM_PASSWORD=very_secure_password \
  -e THREADS=6 \
  -e POSTGRES_MAX_WAL_SIZE=3GB \
  -e IMPORT_STYLE=admin \
  -p 8080:8080 \
  --name nominatim \
  nominatim
```

Reply:
Thanks for the explanation. I missed that part in  the documentation.
The image seems to build but I get a new error.

```bash
Step 27/28 : ENV THREADS=16
 ---> Running in d495ac2015d7
Removing intermediate container d495ac2015d7
 ---> 697e6ca391da
Step 28/28 : CMD /app/start.sh
 ---> Running in 13f588dc0156
Removing intermediate container 13f588dc0156
 ---> 287251ff5735
Successfully built 287251ff5735
Successfully tagged nominatim:latest
/bin/sh: 1: /app/start.sh: Permission denied
```

I cleaned all the containers and images to be sure to run the built one.

The code I used this time:

```bash
#Stop the nominatim container
docker stop nominatim

# clean stopped containers
docker container prune

# change directory to docker folder
cd ~/nominatim-docker/3.6/

docker build -t nominatim . && \
docker run -it --rm --shm-size=8G \
  -e PBF_URL=https://download.geofabrik.de/europe/monaco-latest.osm.pbf \
  -e REPLICATION_URL=https://download.geofabrik.de/europe/monaco-updates/ \
  -e IMPORT_WIKIPEDIA=false \
  -e NOMINATIM_PASSWORD=very_secure_password \
  -e THREADS=6 \
  -e POSTGRES_MAX_WAL_SIZE=3GB \
  -e IMPORT_STYLE=admin \
  -p 8080:8080 \
  --name nominatim \
  nominatim
```

The error does not append if I use `mediagis/nominatim:3.6`

Reply:
Looks like app.sh is not executable. Can you make sure that it is with
```
chmod +x app.sh
```
and rebuild?

Reply:
You could also let the CI test your changes by adding another test to `ci.yml`.

A meaningful assertion would be a bit more work though. Perhaps a check that the line is written to the configuration is enough.

Reply:
@leonardehrenfried I'll look at add another test. Thanks for the suggestion.